lib/ts/recipe/passwordless/index.ts

```diff
@@ -41,39 +41,83 @@ export default class RecipeWrapper {
         flowType: PasswordlessFlowType;
         fetchResponse: Response;
     }> {
-        return Recipe.getInstanceOrThrow().recipeImplementation.createCode({
+        const recipe: Recipe = Recipe.getInstanceOrThrow();
+        const normalisedUserContext = getNormalisedUserContext(input.userContext);
+
+        const createCodeResponse = await recipe.recipeImplementation.createCode({
             ...input,
-            userContext: getNormalisedUserContext(input.userContext),
+            userContext: normalisedUserContext,
         });
+
+        await recipe.recipeImplementation.setLoginAttemptInfo({
+            attemptInfo: {
+                deviceId: createCodeResponse.deviceId,
+                preAuthSessionId: createCodeResponse.preAuthSessionId,
+                flowType: createCodeResponse.flowType,
+            },
+            userContext: normalisedUserContext,
+        });
+
+        return createCodeResponse;
     }
 
-    static resendCode(input: {
-        deviceId: string;
-        preAuthSessionId: string;
-        userContext?: any;
-        options?: RecipeFunctionOptions;
-    }): Promise<{
+    static async resendCodeAndUpdateState(input: { userContext?: any; options?: RecipeFunctionOptions }): Promise<{
         status: "OK" | "RESTART_FLOW_ERROR";
         fetchResponse: Response;
     }> {
-        return Recipe.getInstanceOrThrow().recipeImplementation.resendCode({
+        const recipe: Recipe = Recipe.getInstanceOrThrow();
+        const normalisedUserContext = getNormalisedUserContext(input.userContext);
+
+        const previousAttemptInfo = await recipe.recipeImplementation.getLoginAttemptInfo({
+            userContext: normalisedUserContext,
+        });
+
+        if (previousAttemptInfo === undefined) {
+            throw new Error(
+                "No information for the previous attempt found, " +
+                    "createCode must be called once before trying to resend"
+            );
+        }
+
+        const resendCodeResponse = await recipe.recipeImplementation.resendCode({
             ...input,
-            userContext: getNormalisedUserContext(input.userContext),
+            userContext: normalisedUserContext,
+            deviceId: previousAttemptInfo.deviceId,
+            preAuthSessionId: previousAttemptInfo.preAuthSessionId,
         });
+
+        if (resendCodeResponse.status === "OK") {
+            const currentLoginAttemptInfo = await recipe.recipeImplementation.getLoginAttemptInfo({
+                userContext: normalisedUserContext,
+            });
+
+            if (
+                currentLoginAttemptInfo !== undefined &&
```

Comment:
What about the case when `currentLoginAttemptInfo === undefined`?

Reply:
This check was only relevant if web-js was adding a `lastResend` property to storage for convenience. Since we decided not to do that the entire block was redundant and has been removed